docs/plugins/ingest.asciidoc

```diff
@@ -101,9 +87,7 @@ Throws error when the field is not an array.
 --------------------------------------------------
 {
   "join": {
-    "fields": {
-      "joined_array_field": "other_array_field"
-    }
+    "joined_array_field": "-"
```

Comment:
so,

```
"join" : {
  "field" : "joined_array_field",
  "separator" : "-"
}
```

I think you are correct. it will be more consistent. then it also gets rid of the whole "multiple fields" provided issue. where there is a concrete set of keys to be expected